src/components/node-list/index.js

```diff
@@ -207,6 +253,52 @@ const NodeListProvider = ({
       onToggleNodeSelected(null);
     }
   };
+
+  // Updates the URL parameters when the filter is reset.
+  const updateUrlParamsOnResetFilter = () => {
+    const searchParams = new URLSearchParams(window.location.search);
+    searchParams.delete(params.tags);
+    searchParams.set(params.types, `${NODE_TYPES.task},${NODE_TYPES.data}`);
+
+    history.push(
+      decodeURIComponent(
+        `${window.location.pathname}?${searchParams.toString()}`
+      )
+    );
+  };
+
+  // Reset applied filters to default
+  const onResetFilter = () => {
+    onToggleTypeDisabled({ task: false, data: false, parameters: true });
+    onToggleTagFilter(
+      tags.map((item) => item.id),
+      false
+    );
+
+    updateUrlParamsOnResetFilter();
+  };
+
+  // Updates the reset filter button status based on the node types and tags.
+  useEffect(() => {
```

Comment:
I was thinking maybe we can simplify this 
setIsResetFilterActive(false) --> whenever the 'Reset' button is clicked 

setIsResetFilterActive(true) --> whenever any tag, or filter is clicked. 

So we avoid the complex logic below.

Reply:
I think doing that will work if someone uses the UI to do filtering, but if someone directly changes the url, the reset button status should also watch out for the url.

Reply:
another suggestion is to follow the KISS principal and keep the `Reset` button always on but since this is also design suggestion looping @stephkaiser

Reply:
@ravi-kumar-pilla @rashidakanchwala @stephkaiser  I understand that the logic to set correct reset button status is complex, but are we keeping it simple like always ON or keeping what I did ?

Reply:
My suggestion is keeping it permanently enabled. Even when both the 'Reset' and reset filter are active, it doesn't alter the outcome. Many UI designs follow this convention, where the clear button remains visible even when the form is already empty. That's just my perspective, but would like to hear what others think.

Reply:
I think this is more of a design choice. It might confuse some users if nothing happens on reset, but considering the complexity of checks needed, it should be fine to have it active always.

Reply:
In case if we keep always ON,
If the user clicks the reset button while the filter is already in the default state, the state will still be set to the same values. In this scenario, if the user is zoomed in on the flowchart, it will also be reset.

Reply:
Hi team, thanks for flagging to me. I think its confusing to users if the Reset button is always there even after they press it, theres no way for uses to tell they are on the default/reset view or the customised filtered view unless they look at the url. I think the convention is that when the reset/clear button is visible that implies that changes have been made and it can be reset to default (like when you've applied a filter on a website, a clear button will appear). 

I wanted to ask what the default view is? because in my opinion it should be without any filters applied (nodes, datasets, parameters). Otherwise it's a bit confusing as it implies to users that a filtered view is the default, unless we want to make a filtered view the default then the Reset button will not be visible when those filters are applied. 

What do you all think?

Reply:
The default view is set to 'Nodes', with 'Datasets' enabled and 'Parameters' disabled. Hence, we use 'Reset' instead of 'Clear' filters. 

Since majority including design agree having 'Reset' enabled all the time might confuse users, I think maintaining the way as Jitendra had previously done seems ok. We can see how we can make it less complex.

Reply:
sounds good. For me this is similar to a subscriber pattern. The Reset button status should subscribe to any changes in the query params. I am not sure of the technical feasibility in this ticket, but yes we can modify in future if need be.

Reply:
Another thought - we also shouldn't rely on a colour change in the Reset text to indicate that a change has been applied as this doesn't work with colourblind users. Design fixes for the Reset button below:

Design fixes for the Reset button below:
- font size 13 (same as the Types and Tags label under Filters)
- font colour default FFFFFF 55%, and on hover FFFFFF 85% (similar pattern for Search)
- no underline as it's not a link
- after user clicks on Reset, button should disappear and only reappear if user changes have been made